### ISSUE TITLE
dart2js: Compiler phases no longer show up in observatory

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#26398]
It used to be that compiler phases were available as user tage in observatory cpu profiles when compiling with -v.
Sometime in the past couple of weeks this stopped working.

**Comment by rakudrama** [comment]
@peter-ahe-google Do you know if your change did this?

**Comment by peter-ahe-google** [comment]
Yes, I removed the use of UserTag (see dart:developer). It's relatively easy to add back, but be careful that measuring adds significant overhead. Also, there's a hard-coded limit on the number of UserTags supported in an application (64). This makes it problematic to use UserTags in production code.

**Comment by rakudrama** [comment]
@peter-ahe-google I'd like to see the user tags come back (under a flag if they really are that expensive.)
They are critical to understanding cpu profiles of multi-phase programs.

@iposva-google Talking of overhead, the innocuous looking map literal `{measurer: this},` accounts for **2%** of dart2js total run time with -v.
https://github.com/dart-lang/sdk/blob/master/pkg/compiler/lib/src/common/tasks.dart#L149
This is a known issue - https://github.com/dart-lang/sdk/issues/26046 @fsc8000, how is it going?

@johnmccutchan The cpu profile annotated source text is misleading. In my profile 0.24% (inclusive) is attributed to the line with the map literal, and 2.60% (inclusive) on the next line.  Looking at the disassembly, 1.96% (inclusive) is attributed to the return address of the call to `_InternalLinkedHashMap._InternalLinkedHashMap`, a `pop rcx` instruction that is within the inlining region `Map.Map._fromLiteral@915557746 -> LinkedHashMap.LinkedHashMap`.
I would not expect any part of the map literal to be attributed to a source position after the closing brace.

**Comment by peter-ahe-google** [comment]
I'm sorry I wasn't clear, it's not UserTag that adds overhead, it's the --verbose option that adds overhead and this skews your results. We can't use UserTag without --verbose, but the numbers we get are skewed so UserTag doesn't seem like a good approach.

I'm hoping to get access to the profiler ticks somehow, and use this to create a tool similar to PerfAnal, this should give us exactly the profiling information we need.

**Comment by rakudrama** [comment]
Could we put all the --verbose overhead under a new UserTag?
Then we could ignore the overhead when looking at profiles in observatory.
For example, the 2% for allocating `{measurer: this}` would then be separable from the cost of other map allocations.
